File: Student-Unity/src/main/java/csed/swe/studentunity/model/Task.java
package csed.swe.studentunity.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.net.URL;
import java.util.Date;

@Getter
@Setter
@Entity
public class Task implements Serializable {

    @Id @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "task_id")
    private Long taskId;

    @Column(columnDefinition = "varchar(20)", name = "title")
    private String title;

    @ManyToOne
    @JoinColumn(name = "course_id", referencedColumnName = "course_id")
    private Course course;

    @Column(columnDefinition = "Date")
    @JsonFormat(pattern="yyyy-MM-dd'T'HH:mm:ss")
    private Date dueDate;

    @Column(columnDefinition = "varchar(50)", name = "telegram_link")
    private String telegramLink;

    public Task() {}

    public Task(Long taskId) {
        this.taskId = taskId;
    }

    public Task(String title, Date dueDate, String telegramLink) {
        this.title = title;
        this.dueDate = dueDate;
        this.telegramLink = telegramLink;
    }

    public Task(Long taskId, String title, Course course, Date dueDate, String telegramLink) {
        this.taskId = taskId;
        this.title = title;
        this.course = course;
        this.dueDate = dueDate;
        this.telegramLink = telegramLink;
    }

    public Task(String title, Course course, Date dueDate) {
    }
}

File: Student-Unity-frontend/src/app/models/Task.ts
export class Task {
    taskId: number;
    title?: string;
    courseCode?: string;
    course?: any;
    dueDate?: string;
    telegramLink?: string;
    status?: boolean;

    constructor(
        taskId?: number,
        courseCode?: string,
        title?: string,
        dueDate?: string,
        telegramLink?: string,
        status?: boolean
    ) {
        if (taskId) {
            this.taskId = taskId;
            this.title = title;
            this.courseCode = courseCode;
            this.dueDate = dueDate;
            this.telegramLink = telegramLink;
            this.status = status;
        } else {
            this.taskId = 0;
            this.title = '';
            this.courseCode = '';
            this.course = null;
            this.dueDate = '';
            this.telegramLink = '';
            this.status = false;
        }
    }
}
